hibernate-core/src/test/java/org/hibernate/test/ops/SquashUpdateIntoInsertTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.ops;
+
+import static org.junit.Assert.assertEquals;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.event.service.spi.EventListenerRegistry;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.event.spi.PreUpdateEvent;
+import org.hibernate.event.spi.PreUpdateEventListener;
+import org.hibernate.testing.TestForIssue;
+import org.junit.Test;
+
+/**
+ * @author Yanming Zhou
+ */
+@TestForIssue(jiraKey = "HHH-14228")
+public class SquashUpdateIntoInsertTest {
```

Comment:
BTW, Vlad recommends a better testing template in https://in.relation.to/2018/06/04/best-way-write-hibernate-orm-issue-test-case/, which is preferable and it saves lots of unnecessary boilerplate code. Would love it if you migrated to the testing pattern we are more used to. Usually we extend testing class from either `BaseCoreFunctionalTestCase` or `BaseEntityManagerFunctionalTestCase` and only keep from doing this when the testing is related to booting or suchlike.